src/profiler/profiler.cc

```diff
@@ -205,19 +205,14 @@ void Profiler::DumpProfile(bool peform_cleanup) {
   // If aggregate stats aren't enabled, this won't cause a locked instruction
   std::shared_ptr<AggregateStats> ptr_aggregate_stats = aggregate_stats_.get()
                                                         ? aggregate_stats_ : nullptr;
-  bool first_flag = !first_pass && !num_records_emitted_;
   for (uint32_t i = 0; i < dev_num; ++i) {
     DeviceStats &d = profile_stat[i];
     ProfileStat *_opr_stat;
     while (d.opr_exec_stats_->try_dequeue(_opr_stat)) {
       CHECK_NOTNULL(_opr_stat);
       std::unique_ptr<ProfileStat> opr_stat(_opr_stat);  // manage lifecycle
       opr_stat->process_id_ = i;  // lie and set process id to be the device number
-      if (first_flag) {
-        first_flag = false;
-      } else {
-        file << ",\n";
-      }
+      file << ",\n";
```

Comment:
Do we need this ? Also is there a reason not to use std::endl ?

Reply:
There were two new lines here earlier. Using one `\n` is slightly better for performance since we are anyway flushing the buffer with the second new line